examples/expressions/10_expressions_intro.py

```diff
@@ -0,0 +1,299 @@
+)
+products
+
+# %%
+# Note we are getting previews of the output of intermediate results. For
+# example we can see the added ``"total_price"`` column in the output above.
+# The dropdown at the top allows us to check the structure of the pipeline and
+# all the steps it contains.
+#
+# With skrub we do not need to specify a grid of hyperparameters separately
+# from the pipeline. Instead, we can replace a parameter's value with a skrub
+# "choice" which indicates the range of values we would like to consider during
+# hyperparameter selection.
+#
+# Those choices can be nested arbitrarily. They are not restricted to
+# parameters of a scikit-learn estimator, but they can be anything: choosing
+# between different estimators, arguments to function calls, whole sections of
+# the pipeline etc.
+#
+# In-depth information about choices and hyperparameter/model selection is
+# provided in example (TODO add link).
+#
+# Here we build a skrub ``TableVectorizer`` that contains a couple of choices:
+# the type of encoder for high-cardinality categorical or string columns, and
+# the number of components it uses.
+
+# %%
+n = skrub.choose_int(5, 15, log=True, name="n_components")
+encoder = skrub.choose_from(
+    {
+        "MinHash": skrub.MinHashEncoder(n_components=n),
+        "LSA": skrub.StringEncoder(n_components=n),
+    },
+    name="encoder",
+)
+vectorizer = skrub.TableVectorizer(high_cardinality=encoder)
+
+# %%
+# A transformer does not have to apply to the full DataFrame; we can easily
+# restrict it to some columns. Skrub selectors allow specifying which columns
+# in a flexible way, selecting them by name, name pattern, dtype, or other
+# criteria. They can be combined with the same operators as Python sets. Here
+# for example we vectorize all columns except the ``"basket_ID"`` which we will
+# need for joining.
+
+# %%
+from skrub import selectors as s
+
+vectorized_products = products.skb.apply(vectorizer, cols=s.all() - "basket_ID")
+
+# %%
+# Having access to the underlying dataframe's API, we can easily perform the
+# data-wrangling we need, including joins or other operations that involve
+# multiple tables. All those transformations are being implicitly added as
+# steps in our machine-learning pipeline.
+
+# %%
+aggregated_products = vectorized_products.groupby("basket_ID").agg("mean").reset_index()
+baskets = baskets.merge(aggregated_products, left_on="ID", right_on="basket_ID").drop(
+    columns=["ID", "basket_ID"]
+)
+
+# %%
+# Finally, we add a supervised estimator and our pipeline is complete.
+
+# %%
+from sklearn.ensemble import HistGradientBoostingClassifier
+
+hgb = HistGradientBoostingClassifier(
+    learning_rate=skrub.choose_float(0.01, 0.9, log=True, name="learning_rate")
+)
+predictions = baskets.skb.apply(hgb, y=fraud_flags)
+predictions
+
+# %%
+# We can ask for a full report of the pipeline and inspect the results at every
+# step::
+#
+#     predictions.skb.full_report()
```

Comment:
It would be very useful to have the "full_report" appear by default in a notebook/vscode. This is how people work

Reply:
yes I agree. dumping everything in a directory was faster for prototyping. we should move it to a single page and improve the interface, though that will require a bit of time spent on the front-end side. also, there is the challenge of the page becoming quite large if it contains many tablereports